lua/starfall/libs_sh/holograms.lua

```diff
@@ -189,6 +189,22 @@ function holograms_library.create(pos, ang, model, scale)
 	end
 end
 
+--- Removes all holograms created by the calling chip
+-- @return number of holograms removed
+function holograms_library.removeAll()
+	if CLIENT and instance.data.render.isRendering then SF.Throw("Cannot remove while in rendering hook!", 2) end
+	
+	local holos = instance.data.holograms.holos
+	local removedHolos = 0
+	
+	for holo, _ in pairs(holos) do
+		if CLIENT then checkpermission(instance, holo, "hologram.create") end
```

Comment:
It's accessing `instance.data.holograms.holos` directly, not using `getholo` so I think a validity check is required here

Reply:
Holograms are removed from the holos table when they get deleted so I don't think its necessary.